ipykernel/jsonutil.py

```diff
@@ -83,6 +86,13 @@ def encode_images(format_dict):
         if not jpegdata.startswith(JPEG64):
             jpegdata = encodebytes(jpegdata)
         encoded['image/jpeg'] = jpegdata.decode('ascii')
+        
+    gifdata = format_dict.get('image/gif')
+    if isinstance(gifdata, bytes):
+        # make sure we don't double-encode
+        if not gifdata.startswith(GIF) or not gifdata.startswith(GIF89):
```

Comment:
I don't think this is right. For one thing, it's `not`-ing two mutually exclusive conditions, so I think it will always be True overall. But more importantly, this is meant to be a check whether it's *already* base64 encoded - if you look at all the other formats, it's checking against a base64 encoding of the magic number, not the raw magic number.

There's a handy feature of `.startswith()`: you can give it a tuple to check if any of the strings are prefixes. So I think this should be `if not gifdata.startswith((GIF_64, GIF89_64))`, where the `_64` variables are base64 encoded versions of the magic numbers.